packages/dashboard/src/components/widgets/widgetDropTarget.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { AssetQuery } from '@iot-app-kit/core';
+import { useDrop } from 'react-dnd';
+import { useDispatch } from 'react-redux';
+import { onUpdateAssetQueryAction } from '../../store/actions/updateAssetQuery';
+import { Widget, AppKitWidget } from '../../types';
+import { ItemTypes } from '../dragLayer/itemTypes';
+
+export type WidgetDropTargetProps = {
+  widget: Widget;
+  children: React.ReactNode;
+};
+
+export const WidgetDropTarget: React.FC<WidgetDropTargetProps> = ({ widget, children }) => {
+  const dispatch = useDispatch();
+
+  const [, drop] = useDrop(
+    () => ({
+      accept: ItemTypes.ResourceExplorerAssetProperty,
+      drop: ({ queryAssetsParam }: { queryAssetsParam: AssetQuery }) => {
+        const newAssetQueries = queryAssetsParam;
+
+        const appKitWidget = structuredClone(widget) as AppKitWidget;
+        appKitWidget.widgetId = widget.id;
+
+        dispatch(
+          onUpdateAssetQueryAction({
+            // TODO: There is a mismatch between type expectations and what actually results in data being retrieved here; this expects an AssetQuery[] but actually requires an AssetQuery
```

Comment:
What if we are adding two properties from different asset?

Reply:
I agree this is a problem; I want to discuss the best way to handle it from a data structure perspective with the team before forging ahead with a specific solution.